phabricatoremails/template.py

```diff
@@ -0,0 +1,237 @@
+import textwrap
+from datetime import datetime, timezone
+
+import jinja2
+from phabricatoremails.model.common import Reviewer, ReviewerStatus
+from phabricatoremails.model.phabricator import (
+    ReplyContext,
+    CodeContext,
+    DiffLineType,
+    AffectedFileChange,
+)
+from phabricatoremails.model.phabricator_secure import ExistenceChange
+from premailer import Premailer
+
+
+def is_reply(context):
+    return isinstance(context, ReplyContext)
+
+
+def is_code(context):
+    return isinstance(context, CodeContext)
+
+
+def is_has_comments(event):
```

Comment:
What does `is_has` mean? Seems like `has_comments` would be a good name for this instead.

Reply:
Jinja allows custom ["tests"](https://jinja.palletsprojects.com/en/2.11.x/api/#custom-tests) to check if some data meets a constraint that you define.

Importantly, in Jinja syntax, this looks like:
```
{% if event.revision is secure %}
    hello
{% endif}
```

So, if I want to check if an event has comments, then the test name has to match the `is` syntax. I opted for: `{% if event is has_comments %}`.
Then, I followed Jinja convention of naming the function in Python as `is_$test_name()`.

Reply:
I'll have to look into those custom tests. Can we use `is_commented_on` and `is_securely_commented_on` instead?